Userland/Utilities/sql.cpp

```diff
@@ -173,8 +150,72 @@ int main()
         editor.set_prompt(prompt_for_level(open_indents));
     };
 
-    repl();
+    Core::EventLoop loop;
+    auto sql_client = SQL::SQLClient::construct();
+    int the_connection_id;
+
+    auto read_sql = [&]() {
+        do {
+            String piece = read_next_piece();
+            if (!s_keep_running)
+                break;
+            if (piece.is_empty())
+                continue;
+
+            if (piece.starts_with('.')) {
+                handle_command(piece);
+            } else {
+                auto statement_id = sql_client->sql_statement(the_connection_id, piece);
+                sql_client->async_statement_execute(statement_id);
+                return;
+            }
+        } while (s_keep_running);
+        sql_client->async_disconnect(the_connection_id);
+    };
+
+    sql_client->on_connected = [&](int connection_id) {
+        the_connection_id = connection_id;
+        read_sql();
+    };
+
+    sql_client->on_executed_update = [&](int, int updated, int created, int deleted) {
+        outln("{} row(s) updated, {} created, {} deleted", updated, created, deleted);
+        read_sql();
+    };
+
+    sql_client->on_next_result = [&](int, Vector<String> const& row) {
+        bool first = true;
+        for (auto& column : row) {
+            if (first)
```

Comment:
Was this supposed to be `if (!first)`?

Reply:
You are, of course, right. At this moment I don't have all the machinery in place to test statements that actually return data. Working on that now.